### ISSUE TITLE
Improve merge speed in proofreading tool by reducing roundtrips

### ISSUE THREAD

**Issue by philippotto** [scalableminds/webknossos#7969]
Currently, a merge operation:

- blocks the user by entering a busy state
- adapts the mapping locally (the rendered data is updated directly)
- ensures a saved state (so that later mapping-requests provide correct values)
- unblocks the user

Multiple ideas to improve this:

# Idea 1: Don't ensure a saved state

- when merging two ids A and B (e.g., so that B is changed to A) one can adapt the results of mapping-requests (that the server sent without nowing about the merge) by reassigning B to A
- ensuring a saved state would not be necessary anymore then
- however, split operations would need a new synchronization point (before and after the split; previously it was only after the split). the before-sync is only needed when a merge wasn't saved yet.

# Idea 2: Don't block the user while ensuring a saved state
- the user should be free to move around and inspect stuff. nothing bad should happen when only a save is pending.
- if the user does a split-operation after a merge, the merge needs to be awaited before the split operation is sent to the server, though. thus, we would defer the synchronization point.

Both ideas introduce additional complexity. I'd prefer doing (1) for now, since I have a bad feeling about dissolving the busy-blocking.

**Comment by hotzenklotz** [comment]
@philippotto Is this issue completed with the recent [magic mappings PR](https://github.com/scalableminds/webknossos/pull/7654)?

**Comment by philippotto** [comment]
no, unfortunately not. this issue was created to suggest further improvements for the "magic mappings" performance. probably not high-pri for now, but it could definitely make the merge operation more snappy (think old merger-mode style).
